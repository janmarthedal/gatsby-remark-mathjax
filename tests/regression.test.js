const Remark = require(`remark`)
const plugin = require(`../index`)

describe(`remark mathjax plugin`, () => {
  it(`renders inlineMath node properly`, async () => {
    const equation = `$a^2 + b^2 = c^2$`
    let remark = new Remark()
    for (let parserPlugins of plugin.setParserPlugins()) {
      remark = remark.use(parserPlugins)
    }
    const markdownAST = remark.parse(equation)
    await plugin({ markdownAST })
    expect(markdownAST).toMatchSnapshot()
  })

  it(`renders double $ inlineMath node properly`, async () => {
    const equation = `$$a^2 + b^2 = c^2$$`
    let remark = new Remark()
    for (let parserPlugins of plugin.setParserPlugins()) {
      remark = remark.use(parserPlugins)
    }
    const markdownAST = remark.parse(equation)
    await plugin({ markdownAST })
    expect(markdownAST).toMatchSnapshot()
  })

  it(`renders math node properly`, async () => {
    const equation = `$$\na^2 + b^2 = c^2\n$$`
    let remark = new Remark()
    for (let parserPlugins of plugin.setParserPlugins()) {
      remark = remark.use(parserPlugins)
    }
    const markdownAST = remark.parse(equation)
    await plugin({ markdownAST })
    expect(markdownAST).toMatchSnapshot()
  })
})
