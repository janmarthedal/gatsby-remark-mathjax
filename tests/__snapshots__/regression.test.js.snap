// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`remark mathjax plugin renders double $ inlineMath node properly 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "data": Object {
            "hChildren": Array [
              Object {
                "type": "text",
                "value": "a^2 + b^2 = c^2",
              },
            ],
            "hName": "span",
            "hProperties": Object {
              "className": "inlineMath",
            },
          },
          "position": Position {
            "end": Object {
              "column": 20,
              "line": 1,
              "offset": 19,
            },
            "indent": Array [],
            "start": Object {
              "column": 1,
              "line": 1,
              "offset": 0,
            },
          },
          "type": "html",
          "value": "<span class=\\"mjx-chtml\\"><span class=\\"mjx-math\\" aria-label=\\"a^2 + b^2 = c^2\\"><span class=\\"mjx-mrow\\" aria-hidden=\\"true\\"><span class=\\"mjx-msubsup\\"><span class=\\"mjx-base\\"><span class=\\"mjx-mi\\"><span class=\\"mjx-char MJXc-TeX-math-I\\" style=\\"padding-top: 0.225em; padding-bottom: 0.298em;\\">a</span></span></span><span class=\\"mjx-sup\\" style=\\"font-size: 70.7%; vertical-align: 0.513em; padding-left: 0px; padding-right: 0.071em;\\"><span class=\\"mjx-mn\\" style=\\"\\"><span class=\\"mjx-char MJXc-TeX-main-R\\" style=\\"padding-top: 0.372em; padding-bottom: 0.372em;\\">2</span></span></span></span><span class=\\"mjx-mo MJXc-space2\\"><span class=\\"mjx-char MJXc-TeX-main-R\\" style=\\"padding-top: 0.298em; padding-bottom: 0.446em;\\">+</span></span><span class=\\"mjx-msubsup MJXc-space2\\"><span class=\\"mjx-base\\"><span class=\\"mjx-mi\\"><span class=\\"mjx-char MJXc-TeX-math-I\\" style=\\"padding-top: 0.446em; padding-bottom: 0.298em;\\">b</span></span></span><span class=\\"mjx-sup\\" style=\\"font-size: 70.7%; vertical-align: 0.513em; padding-left: 0px; padding-right: 0.071em;\\"><span class=\\"mjx-mn\\" style=\\"\\"><span class=\\"mjx-char MJXc-TeX-main-R\\" style=\\"padding-top: 0.372em; padding-bottom: 0.372em;\\">2</span></span></span></span><span class=\\"mjx-mo MJXc-space3\\"><span class=\\"mjx-char MJXc-TeX-main-R\\" style=\\"padding-top: 0.077em; padding-bottom: 0.298em;\\">=</span></span><span class=\\"mjx-msubsup MJXc-space3\\"><span class=\\"mjx-base\\"><span class=\\"mjx-mi\\"><span class=\\"mjx-char MJXc-TeX-math-I\\" style=\\"padding-top: 0.225em; padding-bottom: 0.298em;\\">c</span></span></span><span class=\\"mjx-sup\\" style=\\"font-size: 70.7%; vertical-align: 0.513em; padding-left: 0px; padding-right: 0.071em;\\"><span class=\\"mjx-mn\\" style=\\"\\"><span class=\\"mjx-char MJXc-TeX-main-R\\" style=\\"padding-top: 0.372em; padding-bottom: 0.372em;\\">2</span></span></span></span></span></span></span>",
        },
      ],
      "position": Position {
        "end": Object {
          "column": 20,
          "line": 1,
          "offset": 19,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "paragraph",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 20,
      "line": 1,
      "offset": 19,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`remark mathjax plugin renders inlineMath node properly 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "data": Object {
            "hChildren": Array [
              Object {
                "type": "text",
                "value": "a^2 + b^2 = c^2",
              },
            ],
            "hName": "span",
            "hProperties": Object {
              "className": "inlineMath",
            },
          },
          "position": Position {
            "end": Object {
              "column": 18,
              "line": 1,
              "offset": 17,
            },
            "indent": Array [],
            "start": Object {
              "column": 1,
              "line": 1,
              "offset": 0,
            },
          },
          "type": "html",
          "value": "<span class=\\"mjx-chtml\\"><span class=\\"mjx-math\\" aria-label=\\"a^2 + b^2 = c^2\\"><span class=\\"mjx-mrow\\" aria-hidden=\\"true\\"><span class=\\"mjx-msubsup\\"><span class=\\"mjx-base\\"><span class=\\"mjx-mi\\"><span class=\\"mjx-char MJXc-TeX-math-I\\" style=\\"padding-top: 0.225em; padding-bottom: 0.298em;\\">a</span></span></span><span class=\\"mjx-sup\\" style=\\"font-size: 70.7%; vertical-align: 0.513em; padding-left: 0px; padding-right: 0.071em;\\"><span class=\\"mjx-mn\\" style=\\"\\"><span class=\\"mjx-char MJXc-TeX-main-R\\" style=\\"padding-top: 0.372em; padding-bottom: 0.372em;\\">2</span></span></span></span><span class=\\"mjx-mo MJXc-space2\\"><span class=\\"mjx-char MJXc-TeX-main-R\\" style=\\"padding-top: 0.298em; padding-bottom: 0.446em;\\">+</span></span><span class=\\"mjx-msubsup MJXc-space2\\"><span class=\\"mjx-base\\"><span class=\\"mjx-mi\\"><span class=\\"mjx-char MJXc-TeX-math-I\\" style=\\"padding-top: 0.446em; padding-bottom: 0.298em;\\">b</span></span></span><span class=\\"mjx-sup\\" style=\\"font-size: 70.7%; vertical-align: 0.513em; padding-left: 0px; padding-right: 0.071em;\\"><span class=\\"mjx-mn\\" style=\\"\\"><span class=\\"mjx-char MJXc-TeX-main-R\\" style=\\"padding-top: 0.372em; padding-bottom: 0.372em;\\">2</span></span></span></span><span class=\\"mjx-mo MJXc-space3\\"><span class=\\"mjx-char MJXc-TeX-main-R\\" style=\\"padding-top: 0.077em; padding-bottom: 0.298em;\\">=</span></span><span class=\\"mjx-msubsup MJXc-space3\\"><span class=\\"mjx-base\\"><span class=\\"mjx-mi\\"><span class=\\"mjx-char MJXc-TeX-math-I\\" style=\\"padding-top: 0.225em; padding-bottom: 0.298em;\\">c</span></span></span><span class=\\"mjx-sup\\" style=\\"font-size: 70.7%; vertical-align: 0.513em; padding-left: 0px; padding-right: 0.071em;\\"><span class=\\"mjx-mn\\" style=\\"\\"><span class=\\"mjx-char MJXc-TeX-main-R\\" style=\\"padding-top: 0.372em; padding-bottom: 0.372em;\\">2</span></span></span></span></span></span></span>",
        },
      ],
      "position": Position {
        "end": Object {
          "column": 18,
          "line": 1,
          "offset": 17,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "paragraph",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 18,
      "line": 1,
      "offset": 17,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`remark mathjax plugin renders math node properly 1`] = `
Object {
  "children": Array [
    Object {
      "data": Object {
        "hChildren": Array [
          Object {
            "type": "text",
            "value": "a^2 + b^2 = c^2",
          },
        ],
        "hName": "div",
        "hProperties": Object {
          "className": "math",
        },
      },
      "position": Position {
        "end": Object {
          "column": 3,
          "line": 3,
          "offset": 21,
        },
        "indent": Array [
          1,
          1,
        ],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "html",
      "value": "<span class=\\"mjx-chtml MJXc-display\\" style=\\"text-align: center;\\"><span class=\\"mjx-math\\" aria-label=\\"a^2 + b^2 = c^2\\"><span class=\\"mjx-mrow\\" aria-hidden=\\"true\\"><span class=\\"mjx-msubsup\\"><span class=\\"mjx-base\\"><span class=\\"mjx-mi\\"><span class=\\"mjx-char MJXc-TeX-math-I\\" style=\\"padding-top: 0.225em; padding-bottom: 0.298em;\\">a</span></span></span><span class=\\"mjx-sup\\" style=\\"font-size: 70.7%; vertical-align: 0.584em; padding-left: 0px; padding-right: 0.071em;\\"><span class=\\"mjx-mn\\" style=\\"\\"><span class=\\"mjx-char MJXc-TeX-main-R\\" style=\\"padding-top: 0.372em; padding-bottom: 0.372em;\\">2</span></span></span></span><span class=\\"mjx-mo MJXc-space2\\"><span class=\\"mjx-char MJXc-TeX-main-R\\" style=\\"padding-top: 0.298em; padding-bottom: 0.446em;\\">+</span></span><span class=\\"mjx-msubsup MJXc-space2\\"><span class=\\"mjx-base\\"><span class=\\"mjx-mi\\"><span class=\\"mjx-char MJXc-TeX-math-I\\" style=\\"padding-top: 0.446em; padding-bottom: 0.298em;\\">b</span></span></span><span class=\\"mjx-sup\\" style=\\"font-size: 70.7%; vertical-align: 0.584em; padding-left: 0px; padding-right: 0.071em;\\"><span class=\\"mjx-mn\\" style=\\"\\"><span class=\\"mjx-char MJXc-TeX-main-R\\" style=\\"padding-top: 0.372em; padding-bottom: 0.372em;\\">2</span></span></span></span><span class=\\"mjx-mo MJXc-space3\\"><span class=\\"mjx-char MJXc-TeX-main-R\\" style=\\"padding-top: 0.077em; padding-bottom: 0.298em;\\">=</span></span><span class=\\"mjx-msubsup MJXc-space3\\"><span class=\\"mjx-base\\"><span class=\\"mjx-mi\\"><span class=\\"mjx-char MJXc-TeX-math-I\\" style=\\"padding-top: 0.225em; padding-bottom: 0.298em;\\">c</span></span></span><span class=\\"mjx-sup\\" style=\\"font-size: 70.7%; vertical-align: 0.584em; padding-left: 0px; padding-right: 0.071em;\\"><span class=\\"mjx-mn\\" style=\\"\\"><span class=\\"mjx-char MJXc-TeX-main-R\\" style=\\"padding-top: 0.372em; padding-bottom: 0.372em;\\">2</span></span></span></span></span></span></span>",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 3,
      "line": 3,
      "offset": 21,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;
